docs/resources/cluster.md

```diff
@@ -330,8 +330,8 @@ resource "databricks_cluster" "this" {
 The following options are available:
 
 * `zone_id` - (Required) Identifier for the availability zone/datacenter in which the cluster resides. This string will be of a form like “us-west-2a”. The provided availability zone must be in the same region as the Databricks deployment. For example, “us-west-2a” is not a valid zone ID if the Databricks deployment resides in the “us-east-1” region.
-* `availability` - (Optional) Availability type used for all subsequent nodes past the `first_on_demand` ones. Valid values are `SPOT`, `SPOT_WITH_FALLBACK` and `ON_DEMAND`. Note: If `first_on_demand` is zero, this availability type will be used for the entire cluster.
-* `first_on_demand` - (Optional) The first `first_on_demand` nodes of the cluster will be placed on on-demand instances. If this value is greater than 0, the cluster driver node will be placed on an on-demand instance. If this value is greater than or equal to the current cluster size, all nodes will be placed on on-demand instances. If this value is less than the current cluster size, `first_on_demand` nodes will be placed on on-demand instances, and the remainder will be placed on availability instances. This value does not affect cluster size and cannot be mutated over the lifetime of a cluster.
+* `availability` - (Optional) Availability type used for all subsequent nodes past the `first_on_demand` ones. Valid values are `SPOT`, `SPOT_WITH_FALLBACK` and `ON_DEMAND`. Note: If `first_on_demand` is zero, this availability type will be used for the entire cluster. Default value is `SPOT_WITH_FALLBACK`
```

Comment:
Really, defaults are defined on backend (in contrast to defaults that set by the provider), so they could change at some point of time. Maybe add a note about that?

Reply:
note added

Reply:
As I also mentioned in [the issue](https://github.com/databricks/terraform-provider-databricks/issues/1789), the Terraform provider should be able to solve this by setting a default value in the provider code and send that irregardless if the user specified it or not.

Are you also sure that `first_on_demand = 1`? It is considered best practice, but it seems in our experiments that it is defaulting to `first_on_demand = 0`

Reply:
@fuselessmatt defaults may change on backend, and then it will require changes in provider to match them. Provider avoids that by omitting the not specified values, and accept values returned by backend for missing values